pkg/handlers/ar.go

```diff
@@ -42,14 +42,18 @@ func (h *arHandler) HandleFile(ctx logContext.Context, input fileReader) chan Da
 				} else {
 					panicErr = fmt.Errorf("panic occurred: %v", r)
 				}
-				ctx.Logger().Error(panicErr, "Panic occurred when attempting to open ar archive")
+				dataOrErrChan <- DataOrErr{
+					Err: fmt.Errorf("%w: panic error: %v", ErrProcessingFatal, panicErr),
+				}
 			}
 		}()
 
 		start := time.Now()
 		arReader, err := deb.LoadAr(input)
 		if err != nil {
-			ctx.Logger().Error(err, "Error loading AR file")
+			dataOrErrChan <- DataOrErr{
+				Err: fmt.Errorf("%w: loading AR error: %v", ErrProcessingFatal, err),
```

Comment:
Do we care that we "lose" the original `err` since we're wrapping the sentinel error?

Reply:
We would still maintain the error string though correct? Do you think we should use `errors.Join` here instead?

Reply:
> We would still maintain the error string though correct?

Yeah

> Do you think we should use errors.Join here instead?

I think it would have the same effect. It's probably better to wrap the sentinel, right?

Reply:
I believe so. We need to be able to use `errors.Is` to determine if it's a fatal error.